test/Microsoft.Health.Fhir.Shared.Tests.Integration/Features/Operations/Reindex/ReindexJobTests.cs

```diff
@@ -322,20 +330,97 @@ public async Task GivenNewSearchParamCreatedAfterResourcesToBeIndexed_WhenReinde
+                Assert.Empty(searchResults.Results);
+
+                // Create the query <fhirserver>/Observation?resourceFooCode=<observationId>
+                queryParams = new List<Tuple<string, string>> { new(searchParamCode, observationId) };
+
+                // Check that the new search parameter can be used with a different type of resource
+                searchResults = await _searchService.Value.SearchAsync("Observation", queryParams, CancellationToken.None);
+                Assert.Single(searchResults.Results);
+
+                // Confirm that the search parameter "resourceFoo" isn't marked as unsupported
+                Assert.DoesNotContain(searchResults.UnsupportedSearchParameters, t => t.Item1 == searchParamCode);
+
+                // Create the query <fhirserver>/Observation?resourceFooCode=<nonexistent-id>
+                queryParams = new List<Tuple<string, string>> { new(searchParamCode, "nonexistent-id") };
+
+                // No resources should be returned
+                searchResults = await _searchService.Value.SearchAsync("Observation", queryParams, CancellationToken.None);
+                Assert.Empty(searchResults.Results);
+            }
+            finally
+            {
+                cancellationTokenSource.Cancel();
+
+                _searchParameterDefinitionManager.DeleteSearchParameter(searchParam.ToTypedElement());
+                await _testHelper.DeleteSearchParameterStatusAsync(searchParam.Url, CancellationToken.None);
+
+                await _fixture.DataStore.HardDeleteAsync(samplePatient.Wrapper.ToResourceKey(), CancellationToken.None);
+                await _fixture.DataStore.HardDeleteAsync(sampleObservation.Wrapper.ToResourceKey(), CancellationToken.None);
+            }
+        }
+
         private async Task PerformReindexingOperation(CreateReindexResponse response, OperationStatus operationStatus, CancellationTokenSource cancellationTokenSource)
         {
             Task reindexWorkerTask = _reindexJobWorker.ExecuteAsync(cancellationTokenSource.Token);
             ReindexJobWrapper reindexJobWrapper = await _fhirOperationDataStore.GetReindexJobByIdAsync(response.Job.JobRecord.Id, cancellationTokenSource.Token);
 
             int delayCount = 0;
-            while (reindexJobWrapper.JobRecord.Status != operationStatus && delayCount < 10)
+
+            // Set the delay count limit to be higher here for debugging purposes
+            while (reindexJobWrapper.JobRecord.Status != operationStatus && delayCount < 100)
             {
                 await Task.Delay(1000);
                 delayCount++;
                 reindexJobWrapper = await _fhirOperationDataStore.GetReindexJobByIdAsync(response.Job.JobRecord.Id, cancellationTokenSource.Token);
             }
 
-            Assert.InRange(delayCount, 0, 9);
+            Assert.InRange(delayCount, 0, 39);
```

Comment:
>Assert.InRange(delayCount, 0, 39); [](start = 12, length = 34)

Considering how painful this part is, I would suggest to rewrite it to Assert.Equal(operationStatus,reindexJobWrapper.JobRecord.Status)

I think original intent of this check was to make sure we successfully finished operation and not timeout it.

Reply:
If I understand correctly, you are suggesting that we take out `delayCount` entirely.

In that case, would have `while (reindexJobWrapper.JobRecord.Status != operationStatus)`, so we could simply take out this check, right?

Reply:
Not exactly. Delay count can stay because something can go wrong and we have to wait 60 minutes in tests to fail or whatever is timeout on running container. I saying our while loop has two statements, and instead of checking delay count to make sure we left loop in timely manner, we can check other statement.

Reply:
That's much nicer, thanks 😊 fixed in ba197e5.